File: csharp/Structural/11-Flyweight/Flyweight/JobCategoryMetadata.cs
public sealed class JobCategoryMetadata
{
    public string   Category         { get; }
    public string[] ToolRequirements { get; }
    public string[] SafetyChecklist  { get; }
    public string   RegulatoryNote   { get; }

    public JobCategoryMetadata(
        string   category,
        string[] toolRequirements,
        string[] safetyChecklist,
        string   regulatoryNote)
    {
        Category         = category;
        ToolRequirements = toolRequirements;
        SafetyChecklist  = safetyChecklist;
        RegulatoryNote   = regulatoryNote;
    }

    public void PrintMetadata()
    {
        Console.WriteLine($"  Category  : {Category}");
        Console.WriteLine($"  Tools     : {string.Join(", ", ToolRequirements)}");
        Console.WriteLine($"  Safety    : {string.Join(", ", SafetyChecklist)}");
        Console.WriteLine($"  Regulatory: {RegulatoryNote}");
    }
}

File: typescript/Structural/11-Flyweight/Flyweight/JobCategoryMetadata.ts
export class JobCategoryMetadata {
  constructor(
    readonly category: string,
    readonly toolRequirements: string[],
    readonly safetyChecklist: string[],
    readonly regulatoryNote: string,
  ) {}

  printMetadata(): void {
    console.log(`  Category  : ${this.category}`);
    console.log(`  Tools     : ${this.toolRequirements.join(", ")}`);
    console.log(`  Safety    : ${this.safetyChecklist.join(", ")}`);
    console.log(`  Regulatory: ${this.regulatoryNote}`);
  }
}
